client/app/components/group/_history/history.controller.js

```diff
@@ -1,9 +1,112 @@
 class HistoryController {
-  constructor() {
+  constructor(Store, User) {
     "ngInject";
     Object.assign(this, {
+      Store,
+      User,
+      stores: {},
+      users: {},
+      types: {
+        groups: true,
+        stores: true,
+        pickups: true
+      },
+      filteredData: []
     });
   }
+
+  $onInit(){
+    this.getAllStores();
+    this.getAllUsers();
+    this.updateFilteredData();
+  }
+
+  update(){
```

Comment:
I would try use the angular digest cycle more, e.g. having a function that gets called in the template `getEntries()` and that function returns filtered history data. Advantage: less explicit data management code.